pkg/apis/core/validation/validation_test.go

```diff
@@ -12485,6 +12485,70 @@ func TestValidatePodUpdate(t *testing.T) {
 			err:  "",
 			test: "update pod spec schedulingGates: legal deletion",
 		},
+		{
```

Comment:
add a test where the old has a gate, but the new doesn't; in this case we should allow adding a new nodeSelector